File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int n = nums.size(), count = 0;

        for (int i = 1; i < n; i++) {
            if (nums[i - 1] > nums[i]) {
                count++;
            }
        }

        if((count > 1) || (count == 1 && nums[0] < nums[n - 1])){
            return false;
        }

        return true;
    }
};

File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.cs
public class Solution {
    public bool Check(int[] nums) {
        int n = nums.Length;
        int count = 0;

        for (int i = 0; i < n; i++) {
            if (nums[i] > nums[(i + 1) % n]) {
                count++;
            }
        }

        return count < 2;
    }
}